src/main/java/ui/ConfirmChangesDialog.java

```diff
@@ -0,0 +1,26 @@
+package ui;
+
+import javafx.scene.control.Alert;
+import javafx.scene.control.ButtonType;
+import javafx.stage.Stage;
+
+/**
+ * This class represents a confirm dialog that asks the user whether or not to save the current board before
+ * switching to another board, or creating a new board.
+ */
+public class ConfirmChangesDialog extends Alert {
+
+    private static final String DIALOG_TITLE = "Save Changes?";
+    private static final String DIALOG_HEADER = "There are unsaved changes to your current board.";
+    private static final String DIALOG_CONTENT = "If you choose not to save, all unsaved changes will be discarded.\n\n"
+                                               + "Do you want to save them?";
+
+    public ConfirmChangesDialog(Stage stage) {
+        super(AlertType.CONFIRMATION);
```

Comment:
We will need to merge this whole class to `Util.DialogMessage`. Either use the existing `Warning` dialog (not recommended) or create a new confirmation dialog in that class. We would like any dialog to be available for general use.